tmt/steps/execute/simple.py

```diff
@@ -0,0 +1,133 @@
+
+    A simple 'run.sh' script is run directly on the guest, executes all
+    tests in one batch and checks the journal for the test results.
+    """
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(
+            name='shell.simple', doc=_shell_doc, order=50),
+        tmt.steps.Method(
+            name='beakerlib.simple', doc=_beakerlib_doc, order=50),
+        ]
+
+    # Test runner logs
+    logs = ['stdout.log', 'stderr.log', 'nohup.out', 'results.yaml']
+
+    @classmethod
+    def options(cls, how=None):
+        """ Prepare command line options for given method """
+        options = []
+        if how == 'shell.simple':
+            options.append(click.option(
+                '-s', '--script', metavar='SCRIPT', multiple=True,
+                help='Shell script to be executed as a test.'))
+        return options + super().options(how)
+
+    def show(self):
+        """ Show discover details """
+        super().show(['script'])
+
+    def wake(self):
+        """ Wake up the plugin (override data with command line) """
+
+        # Make sure that 'script' is a list
+        tmt.utils.listify(self.data, keys=['script'])
+
+        # Process command line options, apply defaults
+        for option in ['script']:
+            value = self.opt(option)
+            if value:
+                self.data[option] = value
+
+    def prepare_runner(self):
+        """ Place the runner script to workdir """
+        # Detect location of the runner and copy it to workdir
+        script_path = os.path.join(os.path.dirname(__file__), RUNNER)
+        self.debug(f"Copy '{script_path}' to '{self.step.workdir}'.", level=2)
+        # Nothing more to do in dry mode
+        if self.opt('dry'):
+            return
+        shutil.copy(script_path, self.step.workdir)
+
+        # Push the runner to guests
+        for guest in self.step.plan.provision.guests():
+            guest.push()
+
+    def get_logs(self):
+        """ Get logs contents, also print them to info() """
+        # Pull the workdir changes
+        for guest in self.step.plan.provision.guests():
+            guest.pull()
+        # Check each log, show in verbose mode
+        for log in self.logs:
+            try:
+                output = self.step.read(log)
+                self.verbose(log, output.strip(), 'yellow', level=2)
+            except tmt.utils.FileError:
+                pass
+
+    def remove_logs(self, logs=[]):
+        """ Clean up possible old logs """
+        for log in self.logs:
+            path = os.path.join(self.step.workdir, log)
+            if os.path.exists(path):
+                os.remove(path)
+
+    def go(self):
+        """ Execute available tests """
+        super().go()
+
+        # Prepare the runner and remove logs
+        self.remove_logs()
+        self.prepare_runner()
+
+        # Run the runner
+        try:
+            how = 'beakerlib' if 'beakerlib' in self.get('how') else 'shell'
```

Comment:
Could we define the default 'how' somwhere in the module beginnning, instead of hardocoding it here pls?

Reply:
The default method is `shell` and it's defined [here](https://github.com/psss/tmt/blob/master/tmt/steps/__init__.py#L20). This line only handles full method name `beakerlib.simple` versus short name `beakerlib`. Both are supported from the command line.

Reply:
I see, ok makes sense